### ISSUE TITLE
Login fails after update from 1.25.0 to 1.26.0 - syncing forever ... many rooms

### ISSUE THREAD

**Issue by greinick** [matrix-org/synapse#9264]
### Description

We did an update to our system today from V 1.25.0 to the recent 1.26.0. Since that time no Users can log in anymore.

Still logged in users do not see any problem. Even after a restart of synapse.

### Steps to reproduce

Point the webbrowser (Firefox or safari Mac, or FF Linux) to the element webpage or open the element app on Mac, linux, iOS and point it to the URL.

Enter the login username and password for an AD user.

In the webbrowser I get the message: "_Syncing, If you've joined lots of rooms, this might take a while._"
On the iOS iPad I get a spinning logo.

I'd expect to be logged in an beeing granted access to all my room etc.

<!--
Describe how what happens differs from what you expected.

If you can identify any relevant log snippets from _homeserver.log_, please include
those (please be careful to remove any personal or private data). Please surround them with
``` (three backticks, on a line on their own), so that they are formatted legibly.
-->

### Version information

<!-- IMPORTANT: please answer the following questions, to help us narrow down the problem -->

<!-- Was this issue identified on matrix.org or another homeserver? -->
- **Homeserver**:

matrix.dangermouse.filmakademie.de
<!--
 What version of Synapse is running?

You can find the Synapse version with this command:

$ curl http://localhost:8008/_synapse/admin/v1/server_version

(You may need to replace `localhost:8008` if Synapse is not configured to
listen on that port.)
-->
- **Version**:

{"server_version":"1.26.0","python_version":"3.6.9"}

- **Install method**:
<!-- examples: package manager/git clone/pip  -->

install: apt install matrix-synapse-py3
update: apt upgrade

- **Platform**:
<!--
Tell us about the environment in which your homeserver is operating
distro, hardware, if it's running in a vm/container, etc.
-->
Ubuntu 18.04.5 LTS

Virtual server.


**Note**:

Nevertheless: Amazing project, great Work !!!!  - Kowtow

**Comment by JimmyPesto** [comment]
I can confirm the same problem.

**Comment by thomasgraf** [comment]
I have the problem with version 1.26.0 a new installation on a new home server domain. 
I generated the config with the new version. 

Homeserver: 
matrix.waldkirch-chat.de

Version:
Python: 3.8.5
Synapse: 1.26.0

Install method:
pip install matrix-server

Database:
postgres

OS:
Ubuntu 20.04.2 LTS

The problem occurs on element android clients and webcleints.

**Comment by thomasgraf** [comment]
An update to my problem. The same problem occurs when using 1.25.0.

I downgraded and started always with fresh config and database.

With 1.24.0 the problem was gone.

**Comment by JimmyPesto** [comment]
Thanks for the update!

Is it possible to apply the `Rolling back to v1.25.0 after a failed upgrade` database commands and install 1.24 to go directly backwards to a working state?

**Comment by menschentier** [comment]
Maybe noone needs to revert back to older server versions - I had the same problem and it was solved by correcting a misconfigured homeserver.yaml configuration.

1. My configuration is debian but I either had server_name uncommented by accident or the update uncommented this line. 
Line above says: # This is set in /etc/matrix-synapse/conf.d/server_name.yaml for Debian installations.

2. public_baseurl was https://servername but had to be https://servername:8448. 

After I changed these two settings, logging in on new element devices (including nextcloud client) worked again.

Hope this helps anyone...

Kind regards

Simon

**Comment by kovrom** [comment]
Same behavior. @menschentier suggestion was not applicable/didn't help due to the new installation.
New install of Synapse: 1.26.0 on Ubuntu 20.04 LTS

**Comment by menschentier** [comment]
What I found while diagnosing the error was this here:
https://github.com/vector-im/element-android/issues/983
And actually the above mentioned solution is mentioned there as well. 
 
What I tried before (and what is still active in my current working configuration) is setting filter_timeline_limit down to 25. I'm not much of a developer, but it would make sense as this login process seems to run into a timeout after around 90 seconds (nextcloud app).

**Comment by greinick** [comment]
> What I found while diagnosing the error was this here:
> [vector-im/element-android#983](https://github.com/vector-im/element-android/issues/983)
> And actually the above mentioned solution is mentioned there as well.
> 
> What I tried before (and what is still active in my current working configuration) is setting filter_timeline_limit down to 25. I'm not much of a developer, but it would make sense as this login process seems to run into a timeout after around 90 seconds (nextcloud app).

This did not help on our installation.

**Comment by greinick** [comment]
> Maybe noone needs to revert back to older server versions - I had the same problem and it was solved by correcting a misconfigured homeserver.yaml configuration.
> 
> 1. My configuration is debian but I either had server_name uncommented by accident or the update uncommented this line.
>    Line above says: # This is set in /etc/matrix-synapse/conf.d/server_name.yaml for Debian installations.
> 2. public_baseurl was https://servername but had to be https://servername:8448.
> 
> After I changed these two settings, logging in on new element devices (including nextcloud client) worked again.
> 
> Hope this helps anyone...
> 
> Kind regards
> 
> Simon

This did not help on our installation.

**Comment by richvdh** [comment]
>     2. public_baseurl was https://servername but had to be https://servername:8448.

A change introduced in 1.26.0 is that public_baseurl is now set by default (it defaults to `https://servername`). We thought this would be a harmless change but it's certainly possible that it will break login for installations where that is not correct.

So I do recommend that people double-check their `public_baseurl` setting. It is likely that `https://servername:8448` will not be correct for most people either.

**Comment by kovrom** [comment]
Uncommenting public_baseurl, which is commented out by default, fixes the issue. public_baseurl should be uncommented and set to matrix.domain.com or  whereever your synapse lives.

**Comment by greinick** [comment]
> > ```
> > 2. public_baseurl was https://servername but had to be https://servername:8448.
> > ```
> 
> 
> A change introduced in 1.26.0 is that public_baseurl is now set by default (it defaults to `https://servername`). We thought this would be a harmless change but it's certainly possible that it will break login for installations where that is not correct.
> 
> 
> So I do recommend that people double-check their `public_baseurl` setting. It is likely that `https://servername:8448` will not be correct for most people either.

I double cross checked my public_baseurl setting and now it is pointing to the right URL! Sometimes you have to look twice!

Thx. works for me now.

**Comment by JimmyPesto** [comment]
> > ```
> > 2. public_baseurl was https://servername but had to be https://servername:8448.
> > ```
> 
> A change introduced in 1.26.0 is that public_baseurl is now set by default (it defaults to `https://servername`). We thought this would be a harmless change but it's certainly possible that it will break login for installations where that is not correct.
> 
> So I do recommend that people double-check their `public_baseurl` setting. It is likely that `https://servername:8448` will not be correct for most people either.

Solved the issue for me too. Thanks!

**Comment by aaronraimist** [comment]
I don't think this issue should be closed. Lots of people are running in to this. `UPGRADE.rst` should be updated to mention this change.

**Comment by greinick** [comment]
> I don't think this issue should be closed. Lots of people are running in to this. `UPGRADE.rst` should be updated to mention this change.

Good point on updating the `UPGRADE.rst`. I closed the ticket because the issue as such is solved. I reopen it, so may be someone could update the docs.

**Comment by richvdh** [comment]
I agree, this does need addressing.

The reason it is breaking existing installations is that the response to `/login` includes the URI of the homeserver *if `public_baseurl` is set*. If there is no URI in the `/login` response, clients default to using the same URI as they used for login. Now that `public_baseurl` is always set, that URI is always included - but the default may be wrong, hence clients are being directed to the wrong place.

So, one solution might be to only include the URI in the login response if `public_baseurl` is set *explicitly*, rather than inferred. However, that feels a bit magical, and to be honest having an incorrect `public_baseurl` is probably going to bring you problems with other features later on. So on balance I think I prefer updating the docs.

We would probably need to update the INSTALL.md doc as well as UPGRADE.rst.

**Comment by Nourio** [comment]
This change broke my specific setup as well, but I'm confused as to what I should be adding to public_baseurl. I don't own a domain name and I host a server just for me and small group of friends. My friends access it via my external IP and I access it via my LAN IP.

**Comment by callahad** [comment]
We need to make a decision on this before releasing 1.27.

Obvious options:

- Revert the change
- Add documentation, do nothing else
- Fail loudly if unset, making it required for startup (would also need to verify our setup docs).

**Comment by callahad** [comment]
This was introduced in https://github.com/matrix-org/synapse/pull/9159

**Comment by richvdh** [comment]
> Revert the change

note that a bunch of later PRs have built on top of the change, so this is a somewhat intrusive option.

**Comment by callahad** [comment]
We believe we'll want to move toward making this mandatory in the long term. We will need to determine a graceful way to do that. In the shorter term, we need to find a way to unbreak users in the common case when upgrading.

Regardless, we will retroactively add a note to the upgrade docs.

**Comment by callahad** [comment]
Our current plan is to attempt to revert this and figure out how painful that actually is.

If it turns to be too painful, we'll go with temporarily avoiding well-known generation when public_baseurl is not explicitly set.

**Comment by petrov-adg** [comment]
Same problem. Uncommenting public_baseurl was the solution

**Comment by clokep** [comment]
Should be fixed in v1.27.0rc2.
